### PR TITLE
feat(admin): improved default index pages

### LINKED ISSUE #6825: Nudge admins to use more benefits-oriented home page
## Goal

I'd like to see us improve the quality of out-of-box Elgg sites with regard to new-visitor experience by providing a default homepage that emphasizes benefits instead of latest activity.
## Background

Right now I believe the default page when you install Elgg is just a login form and the activity stream. Custom index lets you put random widgets up there. I don't think either of these helps admins tell visitors what the network is about, why they might want to sign up, or what people are expected to accomplish with the site.

I've been reading a lot of UX articles and research and it's clear to me that you really have to hold users' hands. You have to make it explicitly obvious _why_ they should sign up or what they should be doing on your site. That way they can know whether they need to be there in the first place.

Consider the following examples of front page content for logged out/first-time visitors:
- Facebook
  - "Connect with friends and the world around you on Facebook"
  - 3 features
  - registration form
  - Login is not primary
- Twitter 
  - "Start a conversation, explore your interests, and be in the know."
  - Registration + Login forms
  - Links to download apps
- Pinterest
  - "Join Pinterest to find (and save!) all the things that inspire you."
  - Links to sign up
  - Login is not primary
- Tumblr
  - "Follow the blogs you've been hearing about. Share the things you love."
  - Sign up form front-and-center
  - Login link subtle in lower left

Obviously these are very much social networks whose primary value comes from signing up first, but for a lot of successful services, I'm noticing some common elements:
- Clear and benefits-oriented description of what users can do with the site/app
- Opportunity for users to take action, focused on keeping first-time users (so registration is emphasized above logging in)
- Social proof, if available (number of people who use it. List of reputable businesses that use it, testimonials, etc.)
- More detailed feature list for people who want to know more
## Further reading

http://www.useronboard.com/
## Proposal

We should:
- Prompt admins to think about their value proposition (what benefit are they trying to bring to people who sign up for their service).
  - Install flow?
  - Task in admin area?
- Provide a better default homepage that puts this information front-and-center, along with a clear call-to-action for new visitors. Maybe default to register form instead of login form?
- _Demote_ login as the primary action (but not too much!).

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #6825

---

**Core index page (registration _disabled_):**
image

**Core index page (registration _enabled_):**
image

**Core index page (_logged in_):**
image

---

**Activity index page (registration _disabled_):**
image

**Activity index page (registration _enabled_):**
image

### CODE REVIEW

File: mod/custom_index/views/default/custom_index/content.php [hunk 1]
```diff
@@ -6,79 +6,86 @@
...
+$modules = [];
+
+if ($is_module_enabled('about')) {
+	$about = $plugin->about;
+	if (!empty($about)) {
+		$modules[] = elgg_view_module('featured', '', $about);
+	}
+}
+
+if (!elgg_is_logged_in() && $is_module_enabled('register')) {
```
Comment (jeabakker) [inline mod/custom_index/views/default/custom_index/content.php:32]: and registration allowed?!

File: mod/custom_index/views/default/custom_index/content.php [hunk 2]
```diff
@@ -6,79 +6,86 @@
...
+
+if (!elgg_is_logged_in() && $is_module_enabled('login')) {
+	$modules[] = elgg_view_module('featured', elgg_echo('login'), elgg_view_form('login'));
 }
 
 // a view for plugins to extend
-echo elgg_view('index/lefthandside');
+$lefthandside = elgg_view('index/lefthandside');
+if (!empty($lefthandside)) {
+	$modules[] = elgg_view_module('inline', '', $lefthandside);
```
Comment (jeabakker) [inline mod/custom_index/views/default/custom_index/content.php:43]: wouldn't this possibly cause modules in a module?

File: mod/custom_index/views/default/custom_index/content.php [hunk 3]
```diff
@@ -6,79 +6,86 @@
...
-	<div class="elgg-col elgg-col-1of2 custom-index-col2">
-		<div class="elgg-inner">
-<?php
-// right column
 
 // a view for plugins to extend
-echo elgg_view("index/righthandside");
+$righthandside = elgg_view('index/righthandside');
+if (!empty($righthandside)) {
+	$modules[] = elgg_view_module('inline', '', $righthandside);
```
Comment (jeabakker) [inline mod/custom_index/views/default/custom_index/content.php:59]: wouldn't this possibly cause modules in a module?

File: mod/custom_index/views/default/plugins/custom_index/settings.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+/* @var $custom_index \ElggPlugin */
+$custom_index = elgg_extract('entity', $vars);
```
Comment (jeabakker) [inline mod/custom_index/views/default/plugins/custom_index/settings.php:4]: we usually keep this as `$entity` or maybe `$plugin`

File: mod/custom_index/views/default/plugins/custom_index/settings.php [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+$modules = [
+	'about' => '',
+	'register' => '',
+	'login' => '',
+	'activity' => 'activity',
+	'blog' => 'blog',
+	'bookmarks' => 'bookmarks',
+	'file' => 'file',
+	'groups' => 'groups',
+	'users' => 'users',
```
Comment (jeabakker) [inline mod/custom_index/views/default/plugins/custom_index/settings.php:22]: `users` is not a plugin ;)

**Review by jeabakker — CHANGES_REQUESTED** [review]

File: mod/custom_index/views/default/plugins/custom_index/settings.php (hunk 5 above)
Comment (jdalsem) [reply mod/custom_index/views/default/plugins/custom_index/settings.php:22]: good find

File: mod/custom_index/views/default/plugins/custom_index/settings.php (hunk 4 above)
Comment (jdalsem) [reply mod/custom_index/views/default/plugins/custom_index/settings.php:4]: i know, but later we make a `$plugin` so i did want to make clear which plugin i was talking about here

File: mod/custom_index/views/default/custom_index/content.php (hunk 1 above)
Comment (jdalsem) [reply mod/custom_index/views/default/custom_index/content.php:32]: changed

File: mod/custom_index/views/default/custom_index/content.php (hunk 3 above)
Comment (jdalsem) [reply mod/custom_index/views/default/custom_index/content.php]: changed

File: mod/custom_index/views/default/custom_index/content.php (hunk 2 above)
Comment (jdalsem) [reply mod/custom_index/views/default/custom_index/content.php]: changed

**Review by jeabakker — APPROVED** [review]

**Pull request merged**